### PR TITLE
Add BlockFailedDispenseEvent

### PR DESCRIPTION
**Pull request by MiniDigger**
reviving https://github.com/PaperMC/Paper/pull/3205 since the repo is borked or smth?

also added a way to disable the sound effect

### CODE REVIEW

File: Spigot-Server-Patches/0604-Implemented-BlockFailedDispenseEvent.patch [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+index c239265581972d67c7567d7545d62499ebadd195..08f16fe32c12cb22ee5d793ac40b83c9398faf74 100644
+--- a/src/main/java/net/minecraft/server/BlockDispenser.java
++++ b/src/main/java/net/minecraft/server/BlockDispenser.java
+@@ -48,7 +48,14 @@ public class BlockDispenser extends BlockTileEntity {
+         int i = tileentitydispenser.h();
+ 
+         if (i < 0) {
+-            worldserver.triggerEffect(1001, blockposition, 0);
++            // Paper start - BlockFailedDispenseEvent is called here
++            org.bukkit.block.Block block = worldserver.getWorld().getBlockAt(blockposition.getX(), blockposition.getY(), blockposition.getZ());
```
Comment (yannicklamprecht) [inline Spigot-Server-Patches/0604-Implemented-BlockFailedDispenseEvent.patch]: Would personally move that to a protected method that returns if it should play the effect.

protected boolean triggerBlockFailedDispenseEvent(BlockPosition blockPosition)..

And call it in the BlockDropper as well.

Comment (MiniDigger) [reply Spigot-Server-Patches/0604-Implemented-BlockFailedDispenseEvent.patch]: yeah right, this should go into CraftEventFactory, will do that later

**Review by zachbr — DISMISSED** [review]

**Comment by Proximyst** [comment]
@limbo-app ready

**Review by limbo-app — DISMISSED** [review]
Pull request automatically approved for merging - a core team member has previously approved this pull request.

**Comment by Proximyst** [comment]
remember to rebuild your updates, proxi smh my head

@limbo-app ready

**Review by limbo-app — APPROVED** [review]
Pull request automatically approved for merging - a core team member has previously approved this pull request.

**Pull request merged**